src/components/details.js

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react';
+import {
+  Icon,
+  Transition,
+  Message,
+} from 'semantic-ui-react';
+
+
+class HeadingWithDetail extends Component {
+  state = { showDetails: false };
+
+  toggleDetails = () => {
+    if (this.state.showDetails) {
+      this.setState({ showDetails: false });
+    } else {
+      this.setState({ showDetails: true });
+    }
```

Comment:
This isn't always safe. I would suggest using the `setState(updater)` pattern to guarantee you are working with the proper version of `state`.

```js
this.setState(prevState => ({ showDetails: !prevState.showDetails }));
```

Or something similar that conforms with our style guide.